app/jobs/payout/concerns/retries_from_last_enqueued_publisher.rb

```diff
@@ -0,0 +1,42 @@
+module RetriesFromLastEnqueuedPublisher
+  extend ActiveSupport::Concern
+
+  def enqueue_retryable_potential_payout!
```

Comment:
👏  for making the 3 classes 1. Nice reduction in code. Going to a concern here is a problem as it spreads variables across files and makes it harder to reason about. Please extract this to a service class to handle this logic.

Reply:
Sure, I'll move it to a Service.